htsinfer/motif_enrichment.py

```diff
@@ -0,0 +1,100 @@
+"""Calculate motif enrichment from sample data."""
+
+import numpy as np
+from scipy.stats.distributions import binom
+from typing import (Dict, List)
+
+FOREGROUND_DICT = {"UGAUUC": 5, "UAAACC": 3, "AAGUUACCU": 1,
+                   "AAGCCUU": 1, "AGUUCUA": 1, "UUUCCCG": 5}
+BACKGROUND_DICT = {"UGAUUC": 3, "UAAACC": 5, "AAGCCUUAU": 1,
+                   "AGUUCUA": 1, "UUUCCCG": 5, "UUGGAA": 7}
+
+
+def motif_enrichment(
+    foreground: Dict[str, int],
+    background: Dict[str, int]) -> Dict: 
+    """Calculates enrichment and p-values of motifs with similar lengths.
+
+    Args:
+        foreground: dictionary of motifs with counts
+        background: dictionary of motifs with counts
+
+    Returns:
+        Dictionary including motifs, sum of counts of motifs with
+        similar lenght, enrichment score and p-value
+
+    Example:
+        foreground = {"TGATTC": 5, "TAAACC": 3, "AAGTTACCT": 1,
+        "AAGCCTT": 1, "AGTTCTA": 1, "TTTCCCG": 5}
+        background = {"TGATTC": 3, "TAAACC": 5, "AAGCCTTAT": 1,
+        "AGTTCTA": 1, "TTTCCCG": 5, "TTGGAA": 7}
+        >>> motif_enrichment(foreground, background)
+            {'TGATTC': [0.7333333333333334, 0.01646090534979423],
+            'TAAACC': [0.7333333333333334, 0.21947873799725662],
+            'AAGTTACCT': [1.375, 0.26337448559670795],
+            'AAGCCTT': [1.6041666666666667, 0.26337448559670795],
+            'AGTTCTA': [1.6041666666666667, 0.26337448559670795],
+            'TTTCCCG': [1.6041666666666667, 0.01646090534979423]}
+    """
+    # Foreground dictionary
+    new_dict_foreground: Dict = dict()
+    new_dict_foreground_prob: Dict = dict()
+    new_dict_foreground_final: Dict = dict()
+    length_list_f: List = list()
+
+    # Background dictionary
+    new_dict_background: Dict = dict()
+    new_dict_background_prob: Dict = dict()
+    length_list_b: List = list()
+
+    # Foreground dictionary
+    # Calulates sum of motifs of same length
+    for i in list(foreground.keys()):
+
+        if len(i) in length_list_f:
+            new_dict_foreground[len(i)] += foreground[i]
+        else:
+            new_dict_foreground[len(i)] = foreground[i]
+            length_list_f.append(len(i))
+
+    # Calculates probability of occurrence of motifs
```

Comment:
This calculates the frequency of occurrence of motifs of a given length among all lengths. Instead, it should calculate the frequency of occurrence of one particular motif among all motifs of that specific length. Using the example at the top, among motifs of length 6 we have in the foreground:
"UGAUUC": 5, "UAAACC": 3
and in the background
"UGAUUC": 3, "UAAACC": 5, "UUGGAA": 7
To account for the possibility that the foreground and background motif sets are not perfectly overlapping, we would create the foreground and background dictionary such that they contain the same motifs, adding a pseudocount to all motifs before calculating their frequencies. In the example above, we would extend the foreground dictionary to also contain UUGGAA, and then, by adding a pseudo-count of 1 to each motif, we would have the following foreground and background dictionaries
"UGAUUC": 6, "UAAACC": 4, "UUGGAA": 1
and in the background
"UGAUUC": 4, "UAAACC": 6, "UUGGAA": 8
Then the frequency of UGAUUC in foreground would be 6/11, in background 4/18, enrichment value would be 6*18/(4*11) and the p value will be calculated based on 6 out of 11 occurrences assuming a probability of 4/18.

Reply:
Thanks for the feedback. 
I'll try and implement it and push the changes once their ready.

Reply:
Given your example, do I need to increase the pseudocode for all motifs of the same length in the dictionary or just of the ones that were included in the respective dictionary from the beginning? Just in case more than one motif of the same length needs to be added to the same dictionary. If we would add another motif, say "UGCUGC" to foreground, would the foreground look like this: "UGAUUC": 7, "UAAACC": 5, "UUGGAA": 2, "UGCUGC": 1 or like this: "UGAUUC": 7, "UAAACC": 5, "UUGGAA": 1, "UGCUGC": 1.

Reply:
You would need to add the same pseudocount to all motifs